kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +275,127 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
     });
   }
 
+  @Test
+  public void testConfigurationContainsUnknownCluster() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "bad_cluster";
+    when(burrow.getClusterBootstrapServers(clusterName)).thenThrow(new IOException());
+    Map<String, Object> conf = mockConfForCluster(
+            clusterName, "l1.example.com:9092", "l2.example.com:9092"
+    );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    freshness.burrow = burrow;
+
+    Optional<String> msg = freshness.validateClusterConf(conf);
+    Assert.assertTrue(msg.isPresent());
+    Assert.assertTrue(msg.get().contains("failed to read cluster detail from Burrow: "));
+    Assert.assertEquals(1.0,
+            freshness.getMetricsForTesting().burrowClusterDetailReadFailed.labels(clusterName).get(), 0.0);
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsValidServersMatchBurrow() throws Exception {
```

Comment:
This has nothing to do with the validation mode, its just that matching servers don't cause a validation error